src/commands/fun/poll.ts

```diff
@@ -1,111 +1,197 @@
 import { BOT } from '@root/config';
-import { Message, MessageEmbed, MessageReaction } from 'discord.js';
+import { ApplicationCommandOptionData, ButtonInteraction, CommandInteraction, Message, MessageActionRow, MessageButton, MessageEmbed } from 'discord.js';
 import parse from 'parse-duration';
 import { Command } from '@lib/types/Command';
 
-const QUESTION_CHAR_LIMIT = 1024;
-const EMBED_TITLE_CHAR_LIMIT = 256;
+const QUESTION_CHAR_LIMIT = 256;
+const args = ['Single', 'Multiple'];
 
 export default class extends Command {
 
-	description = `Have ${BOT.NAME} create a poll for you`;
-	aliases = ['vote'];
-	usage = '<timespan>|<question>|<choices...>';
-	extendedHelp = 'You can have two to ten choices.';
+	description = `Have ${BOT.NAME} create a poll for you.`;
+
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'timespan',
+			description: `How long your poll should last. Acceptable formats include '5s', '5m', '5h', '5h30m', '7h30m15s'...`,
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'question',
+			description: `What would you like to ask?`,
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'choices',
+			description: `A poll can have 2-10 choices. Separate choices with '|' (no spaces/quotes).`,
```

Comment:
I personally think we should keep list arguments to comma-separated. @renross @bladetyr can weigh in though

Reply:
This would mean if poll options use commas to, e.g. separate items, they would all be split between choices. What if someone wanted to make a poll where the choices were, say, "Eggs, ketchup, and cheese" and "Eggs, sausage, and bacon"? That would result in 6 choices not 2

Reply:
Why not have four required args (question, time, option 1, option 2) and then a bunch of optional args for the rest of the options?

Reply:
I was originally going to do something like that, but the issue that came to mind was if someone (somehow) skipped an optional choice while filling out the rest of their choices. Unless you were thinking of the code parsing each optional choice individually.

Reply:
You could pretty easily filter out options that are null and just show the ones that were given.

Reply:
I like the options idea simply because that's more user friendly (discord tells the user that there are options that can be inputted) rather than someone doing "/poll red|green|blue," but not using options is faster than hitting tab to input poll options. I don't really like the comma split on the off chance that people put 'and' at the end (ex: 'red, green, and blue')

TLDR: I think the poll command is fine as-is, but making them options would be more user-friendly. 'red|green|blue' is faster to input though, so it's a matter of efficiency vs end user ease

Reply:
That's what I was thinking, plus the instructions are clear enough, I'd like to think (saying to separate with |).